exporting/tests/test_exporting_engine.c

```diff
@@ -0,0 +1,401 @@
+{
+    expect_function_call(__wrap_uv_thread_create);
+
+    expect_value(__wrap_uv_thread_create, thread, &engine->connector_root->instance_root->thread);
+    expect_value(__wrap_uv_thread_create, worker, graphite_connector_worker);
+    expect_value(__wrap_uv_thread_create, arg, engine->connector_root->instance_root);
+
+    assert_int_equal(__real_init_connectors(engine), 0);
+}
+
+static void test_exporting_engine(void **state)
+{
+    (void)state;
+
+    struct engine *engine = (struct engine *)malloc(sizeof(struct engine));
+    memset(engine, 0xDB, sizeof(struct engine));
+    engine->after = 1;
+    engine->before = 2;
+
+    expect_function_call(__wrap_read_exporting_config);
+    will_return(__wrap_read_exporting_config, engine);
+
+    expect_function_call(__wrap_init_connectors);
+    expect_memory(__wrap_init_connectors, engine, engine, sizeof(struct engine));
+    will_return(__wrap_init_connectors, 0);
+
+    expect_function_calls(__wrap_now_realtime_sec, 2);
+    will_return(__wrap_now_realtime_sec, 1);
+    will_return(__wrap_now_realtime_sec, 2);
+
+    expect_function_call(__wrap_mark_scheduled_instances);
+    expect_memory(__wrap_mark_scheduled_instances, engine, engine, sizeof(struct engine));
+    will_return(__wrap_mark_scheduled_instances, 0);
+
+    expect_function_call(__wrap_prepare_buffers);
+    expect_memory(__wrap_prepare_buffers, engine, engine, sizeof(struct engine));
+    will_return(__wrap_prepare_buffers, 0);
+
+    expect_function_call(__wrap_notify_workers);
+    expect_memory(__wrap_notify_workers, engine, engine, sizeof(struct engine));
+    will_return(__wrap_notify_workers, 0);
+
+    expect_function_call(__wrap_send_internal_metrics);
+    expect_memory(__wrap_send_internal_metrics, engine, engine, sizeof(struct engine));
+    will_return(__wrap_send_internal_metrics, 0);
+
+    void *ptr = malloc(sizeof(int));
+    assert_ptr_equal(exporting_main(ptr), NULL);
+
+    free(engine);
+    free(ptr);
+}
+
+static void test_prepare_buffers(void **state)
+{
+    (void)state;
+
+    struct engine *engine = (struct engine *)malloc(sizeof(struct engine));
+    memset(engine, 0xD1, sizeof(struct engine));
+    engine->after = 1;
+    engine->before = 2;
+
+    engine->connector_root = (struct connector *)malloc(sizeof(struct connector));
+    struct connector *connector = engine->connector_root;
+    connector->next = NULL;
+    connector->start_batch_formatting = __mock_start_batch_formatting;
+    connector->start_host_formatting = __mock_start_host_formatting;
+    connector->start_chart_formatting = __mock_start_chart_formatting;
+    connector->metric_formatting = __mock_metric_formatting;
+    connector->end_chart_formatting = __mock_end_chart_formatting;
+    connector->end_host_formatting = __mock_end_host_formatting;
+    connector->end_batch_formatting = __mock_end_batch_formatting;
+
+    connector->instance_root = (struct instance *)malloc(sizeof(struct instance));
```

Comment:
Is it not safe to use `callocz` here?

Reply:
It's safe. I'll review/rearrange tests later.